utils/src/treev2/quickPickWizard/quickPickAzureResource/tempTypes.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../../../index';
+import { ContextValueFilterableTreeNode } from '../ContextValueQuickPickStep';
+
+// TODO: THIS FILE IS TEMPORARY //
+// It needs to be replaced by real Resources extension interfaces //
+
+export type ResourceGroupsItem = ContextValueFilterableTreeNode;
+
+export type SubscriptionItem = ResourceGroupsItem & {
+    subscription: ApplicationSubscription;
+};
+
+export type GroupingItem = ResourceGroupsItem & {
+    resourceType?: types.AzExtResourceType
+}
+
+export type AppResourceItem = ResourceGroupsItem & ApplicationResource;
+
+type ResourceBase = {};
+
+/**
+ * Represents an individual resource in Azure.
+ * @remarks The `id` property is expected to be the Azure resource ID.
+ */
+export interface ApplicationResource extends ResourceBase {
+    readonly subscription: ApplicationSubscription;
+    readonly azExtResourceType?: types.AzExtResourceType;
+    readonly resourceGroup?: string;
+}
+
+export type ApplicationSubscription = unknown;
+
+export interface Box<T> {
+    unwrap(): T;

```

Comment:
We originally had it as [`unwrap<T>(): Promise<T>`](https://github.com/microsoft/vscode-azuretools/blob/e7e66a25e7f3f21b2e08cbfbb196a90c76ca65fb/utils/hostapi.v2.d.ts#L15) but I think it makes sense to not make it a `Promise`.

Reply:
I feel like I had a reason for doing `Promise<T>` but I can't remember what it is.

Reply:
We need to use unwrap in a lot of non-async places (ex: in `ContextValueQuickPickStep#isDirectPick` so I will make it not a Promise for now.